src/definers.lisp

```diff
@@ -188,3 +188,94 @@
           (let* ((namespace (symbol-namespace ',name))
                  (hash-table (namespace-binding-table namespace)))
             (setf (gethash name hash-table) new-value)))))))
+
+;;; Definer forms
+
+(defun car-or-x (x)
+  (if (consp x)
+      (car x)
+      x))
+
+(defun normalize-arglist (arglist)
+  "Makes sure the argument list contains an &REST parameter for &KEY and
```

Comment:
This, for some reason, looks like a thing that could use `alexandria:parse-ordinary-lambda-list` - did you try it?